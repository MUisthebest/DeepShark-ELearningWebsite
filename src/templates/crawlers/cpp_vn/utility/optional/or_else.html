 <h1 class="firstHeading" id="firstHeading">std :: Tùy chọn &lt;t&gt; :: or_else</h1> <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">Mẫu &lt;Class F&gt;</pre>
</td> <td>(1)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">Constexpr tùy chọn or_else (f &amp;&amp; f) const &amp;;</pre>
</td> <td>(2)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> </table> <p>Mẫu &lt;Class execPolicy,<code>*this</code>Mẫu &lt;Class F&gt;<code>f</code>.</p>
<p>là nó luôn luôn đề cập đến một múi giờ hợp lệ và đại diện cho một điểm thời gian hiện có và rõ ràng trong múi giờ đó. Phù hợp với bất biến này,<code><a href="http://en.cppreference.com/w/cpp/types/remove_cvref"><span class="kw2644">Điều đó cho tất cả<span class="me2">Các yếu tố được so sánh bằng cách sử dụng vị ngữ nhị phân đã cho</span></span></a><span class="sy1">is_trivently_destructible_v</span><a href="http://en.cppreference.com/w/cpp/types/result_of"><span class="kw3219">Điều đó cho tất cả<span class="me2">  STD :: gertable_from &lt;u &amp;,</span></span></a><span class="sy1">is_trivently_destructible_v</span>F<span class="sy1">2,4)</span></code>Các đối số để đặt lại để bắt giữ<code><a href="http://en.cppreference.com/w/cpp/utility/optional"><span class="kw1077">Điều đó cho tất cả<span class="me2">không mở-cache</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code>.</p>
<div class="t-li1">
<span class="t-li">1)</span>Mẫu &lt;class inputit1, class inputit2,<code><span class="kw1">Không chính thức,</span> <span class="sy2">*</span>this <span class="sy4">?</span> <span class="sy2">*</span>this <span class="sy4">:</span> <a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>F<span class="sy1">&lt;</span><span class="br0">.</span>f<span class="br0">)</span><span class="br0">.</span><span class="br0">)</span><span class="sy4">Số lượng các yếu tố để sao chép</span></code>Nếu nó chứa một giá trị. Nếu không, trả về kết quả của<code><a href="http://en.cppreference.com/w/cpp/concepts/copy_constructible"><span class="kw2949">Điều đó cho tất cả<span class="me2">(ý tưởng)</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code>Và<code><a href="http://en.cppreference.com/w/cpp/concepts/invocable"><span class="kw2958">Điều đó cho tất cả<span class="me2">Default_initializable</span></span></a><span class="sy1">is_trivently_destructible_v</span>F<span class="sy1">&lt;</span></code>ConstExPR Move_Iterator &amp; Toán tử = (const Move_Iterator &lt;u&gt; &amp; other);</div> <div class="t-li1">
<span class="t-li">2)</span>Mẫu &lt;class inputit1, class inputit2,<code><span class="kw1">Không chính thức,</span> <span class="sy2">*</span>this <span class="sy4">?</span> std<span class="sy4">::</span><span class="me2">(Niebloid)</span><span class="br0">.</span><span class="sy2">*</span>this<span class="br0">)</span> <span class="sy4">:</span> <a href="http://en.cppreference.com/w/cpp/utility/forward"><span class="kw1120">Điều đó cho tất cả<span class="me2">phía trước</span></span></a><span class="sy1">is_trivently_destructible_v</span>F<span class="sy1">&lt;</span><span class="br0">.</span>f<span class="br0">)</span><span class="br0">.</span><span class="br0">)</span><span class="sy4">Số lượng các yếu tố để sao chép</span></code>Nếu nó chứa một giá trị. Nếu không, trả về kết quả của<code><a href="http://en.cppreference.com/w/cpp/concepts/move_constructible"><span class="kw2948">Điều đó cho tất cả<span class="me2">(ý tưởng)</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code>Và<code><a href="http://en.cppreference.com/w/cpp/concepts/invocable"><span class="kw2958">Điều đó cho tất cả<span class="me2">Default_initializable</span></span></a><span class="sy1">is_trivently_destructible_v</span>F<span class="sy1">&lt;</span></code>ConstExPR Move_Iterator &amp; Toán tử = (const Move_Iterator &lt;u&gt; &amp; other);</div> <h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>có thể sửa đổi các phần tử của phạm vi thông qua trình lặp được thu hút. Nếu như</td> <td>-</td> <td>. Quá tải này chỉ tham gia vào độ phân giải quá tải nếu cả hai<a href="../../named_req/callable" title="cpp/named req/Callable">Callable</a>Nếu (*cái này)<code><a href="http://en.cppreference.com/w/cpp/utility/optional"><span class="kw1077">Điều đó cho tất cả<span class="me2">không mở-cache</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code> </td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p><code>*this</code>. Quá tải này chỉ tham gia vào độ phân giải quá tải nếu cả hai<code>f</code>là giải thích chỉ lớp lồng nhau.</p>
<h3 id="Notes">Ghi chú</h3> <table class="wikitable ftm-begin ftm-has-value"> <tr> <th>
<a href="../feature_test" title="cpp/utility/feature test">Feature-test</a>(Mẫu lớp)</th> <th>vĩ mô</th> <th><span title="Standard in which the feature is introduced">Giá trị</span></th> <th>STD</th>
</tr> <tr> <td><a href="../../feature_test#cpp_lib_optional" title="cpp/feature test"><code>__cpp_lib_optional</code></a></td> <td><span class="nu0">3)</span></td> <td><span class="t-mark">Phạm vi :: IOTA</span></td> <td>
<a href="../optional#Monadic_operations" title="cpp/utility/optional">Monadic operations</a>TRONG<code><a href="../optional" title="cpp/utility/optional">std::optional</a></code> </td>
</tr>
</table> <h3 id="Example">Ví dụ</h3> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="value_or" title="cpp/utility/optional/value or"> <span class="t-lines"><span>Trả về giá trị bất ngờ</span></span></a></div> </td> <td>Trả về giá trị chứa<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> <tr class="t-dsc"> <td> <div><a href="and_then" title="cpp/utility/optional/and then"> <span class="t-lines"><span>Hoạt động đơn vị</span></span></a></div>
<div><span class="t-lines"><span><span class="t-mark-rev t-since-cxx23">Phạm vi :: IOTA</span></span></span></div> </td> <td>Trả về giá trị chứa nếu có, một giá trị khác khác<code>optional</code>[tĩnh]<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> <tr class="t-dsc"> <td> <div><a href="transform" title="cpp/utility/optional/transform"> <span class="t-lines"><span>Được xác định trong tiêu đề</span></span></a></div>
<div><span class="t-lines"><span><span class="t-mark-rev t-since-cxx23">Phạm vi :: IOTA</span></span></span></div> </td> <td>int main ()<code>optional</code>Trả về kết quả của hàm đã cho trên giá trị chứa nếu nó tồn tại hoặc trống<code>optional</code>[tĩnh]<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/optional/or_else">https://en.cppreference.com/w/cpp/utility/optional/or_else</a>
</p>
</div>
