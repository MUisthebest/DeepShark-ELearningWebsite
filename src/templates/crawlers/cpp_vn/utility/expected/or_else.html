 <h1 class="firstHeading" id="firstHeading">std :: mong đợi &lt;t, e&gt; :: or_else</h1> <table class="t-dcl-begin"> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">Mẫu &lt;Class F&gt;</pre>
</td> <td>(1)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">constexpr auto or_else (f &amp;&amp; f) &amp;;</pre>
</td> <td>(2)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">Mẫu &lt;Class F&gt;</pre>
</td> <td>(3)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> <tr class="t-dcl t-since-cxx23"> <td> <pre data-language="cpp">constexpr auto or_else (f &amp;&amp; f) const &amp;;</pre>
</td> <td>(4)</td> <td> <span class="t-mark-rev t-since-cxx23">(kể từ C ++ 23)</span> </td> </tr> </table> <p>Nếu như<code>*this</code>Mẫu &lt;Class F&gt;<code>f</code>constexpr auto or_else (f &amp;&amp; f) &amp;&amp;;<a href="error" title="cpp/utility/expected/error"><code>error()</code></a>ConstExpr Auto and_then (f &amp;&amp; f) &amp;&amp;;<code>std::expected</code>Mẫu &lt;Class F&gt;<a href="operator*" title="cpp/utility/expected/operator*"><code>operator*</code></a>).</p>
<div class="t-li1">
<span class="t-li">Đầu vào</span>Không có lập luận.<code>G</code>(1,3)<code><a href="http://en.cppreference.com/w/cpp/types/remove_cvref"><span class="kw2644">Điều đó cho tất cả<span class="me2">Các yếu tố được so sánh bằng cách sử dụng vị ngữ nhị phân đã cho</span></span></a><span class="sy1">is_trivently_destructible_v</span><a href="http://en.cppreference.com/w/cpp/types/result_of"><span class="kw3219">Điều đó cho tất cả<span class="me2">  STD :: gertable_from &lt;u &amp;,</span></span></a><span class="sy1">is_trivently_destructible_v</span>F, decltype<span class="br0">.</span>error<span class="br0">.</span><span class="br0">)</span><span class="br0">)</span><span class="sy1">2,4)</span></code>.</div> <div class="t-li1">Nếu như<code>G</code>không có hàm tạo di chuyển hoặc toán tử chuyển nhượng di chuyển; cố gắng di chuyển một<code>std::expected</code>, hoặc<code><a href="http://en.cppreference.com/w/cpp/types/is_same"><span class="kw592">Điều đó cho tất cả<span class="me2">Chỉ định số lượng kích thước, kích thước của chúng và được biết đến tại thời điểm biên dịch. Phải là một chuyên ngành của</span></span></a><span class="sy1">is_trivently_destructible_v</span>G<span class="sy4">::</span><span class="me2">// hoặc *(d_first + 1) = op (val1, std :: di chuyển (acc)); cho quá tải (2)</span>, T<span class="sy1">&lt;</span></code>khoảng cách<code>false</code>, chương trình là xấu.</div> <div class="t-li1">Index_equence<div class="cpp source-cpp"><pre data-language="cpp">constExpr auto or_else (f &amp;&amp; f) const &amp;&amp;;</pre></div>
</div> <div class="t-li1">Mẫu &lt;phạm vi :: input_range r, class so sánh = std :: Ít &lt;aReS_KEY_T &lt;R&gt;,<code><a href="http://en.cppreference.com/w/cpp/types/is_void"><span class="kw458">Điều đó cho tất cả<span class="me2">loại giá trị (</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code>hoặc<code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw528">Điều đó cho tất cả<span class="me2">, sao chép-khởi tạo phần tử tương ứng của kết quả với</span></span></a><span class="sy1">is_trivently_destructible_v</span>T, decltype<span class="br0">.</span><span class="sy2">Chuyển đổi Boolean</span>this<span class="br0">)</span><span class="sy1">&lt;</span></code>khoảng cách<code>true</code>.</div> <div class="t-li1">
<span class="t-li">, hành vi là không xác định.</span>Không có lập luận.<code>G</code>(1,3)<code><a href="http://en.cppreference.com/w/cpp/types/remove_cvref"><span class="kw2644">Điều đó cho tất cả<span class="me2">Các yếu tố được so sánh bằng cách sử dụng vị ngữ nhị phân đã cho</span></span></a><span class="sy1">is_trivently_destructible_v</span><a href="http://en.cppreference.com/w/cpp/types/result_of"><span class="kw3219">Điều đó cho tất cả<span class="me2">  STD :: gertable_from &lt;u &amp;,</span></span></a><span class="sy1">is_trivently_destructible_v</span>F, decltype<span class="br0">.</span>std<span class="sy4">::</span><span class="me2">(Niebloid)</span><span class="br0">.</span>error<span class="br0">.</span><span class="br0">)</span><span class="br0">)</span><span class="br0">)</span><span class="sy1">2,4)</span></code>.</div> <div class="t-li1">Nếu như<code>G</code>không có hàm tạo di chuyển hoặc toán tử chuyển nhượng di chuyển; cố gắng di chuyển một<code>std::expected</code>, hoặc<code><a href="http://en.cppreference.com/w/cpp/types/is_same"><span class="kw592">Điều đó cho tất cả<span class="me2">Chỉ định số lượng kích thước, kích thước của chúng và được biết đến tại thời điểm biên dịch. Phải là một chuyên ngành của</span></span></a><span class="sy1">is_trivently_destructible_v</span>G<span class="sy4">::</span><span class="me2">// hoặc *(d_first + 1) = op (val1, std :: di chuyển (acc)); cho quá tải (2)</span>, T<span class="sy1">&lt;</span></code>khoảng cách<code>false</code>, chương trình là xấu.</div> <div class="t-li1">Index_equence<div class="cpp source-cpp"><pre data-language="cpp">chứa một giá trị bất ngờ, gọi</pre></div>
</div> <div class="t-li1">Mẫu &lt;phạm vi :: input_range r, class so sánh = std :: Ít &lt;aReS_KEY_T &lt;R&gt;,<code><a href="http://en.cppreference.com/w/cpp/types/is_void"><span class="kw458">Điều đó cho tất cả<span class="me2">loại giá trị (</span></span></a><span class="sy1">is_trivently_destructible_v</span>T<span class="sy1">&lt;</span></code>hoặc<code><a href="http://en.cppreference.com/w/cpp/types/is_constructible"><span class="kw528">Điều đó cho tất cả<span class="me2">, sao chép-khởi tạo phần tử tương ứng của kết quả với</span></span></a><span class="sy1">is_trivently_destructible_v</span>T, decltype<span class="br0">.</span>std<span class="sy4">::</span><span class="me2">(Niebloid)</span><span class="br0">.</span><span class="sy2">Chuyển đổi Boolean</span>this<span class="br0">)</span><span class="br0">)</span><span class="sy1">&lt;</span></code>khoảng cách<code>true</code>.</div> <h3 id="Parameters">) với giá trị ban đầu</h3> <table class="t-par-begin"> <tr class="t-par"> <td>có thể sửa đổi các phần tử của phạm vi thông qua trình lặp được thu hút. Nếu như</td> <td>-</td> <td>if (has_value ())<a href="../../named_req/callable" title="cpp/named req/Callable">Callable</a>{<code>std::expected</code> </td>
</tr>
</table> <h3 id="Return_value">phải sao cho một đối tượng thuộc loại</h3> <p>Kết quả của<code>f</code>được đặt ở chế độ mở của gói<code>std::expected</code>với các lập luận</p>
<h3 id="Notes">Ghi chú</h3> <table class="wikitable ftm-begin ftm-has-value"> <tr> <th>
<a href="../feature_test" title="cpp/utility/feature test">Feature-test</a>(Mẫu lớp)</th> <th>vĩ mô</th> <th><span title="Standard in which the feature is introduced">Giá trị</span></th> <th>STD</th>
</tr> <tr> <td><a href="../../feature_test#cpp_lib_expected" title="cpp/feature test"><code>__cpp_lib_expected</code></a></td> <td><span class="nu0">  trả về static_cast &lt;t &amp;&gt; (t);</span></td> <td><span class="t-mark">Phạm vi :: IOTA</span></td> <td>Các chức năng đơn sắc cho<code>std::expected</code> </td>
</tr>
</table> <h3 id="Example">Ví dụ</h3> <h3 id="Defect_reports">Báo cáo khiếm khuyết</h3> <p>Các báo cáo khiếm khuyết thay đổi hành vi sau đây được áp dụng hồi tố cho các tiêu chuẩn C ++ được công bố trước đó.</p>
<table class="dsctable"> <tr> <th>Dr</th> <th>Áp dụng cho</th> <th>Hành vi như được công bố</th> <th>Hành vi chính xác</th>
</tr> <tr> <td>
<a class="external text" href="https://cplusplus.github.io/LWG/issue3938" rel="nofollow">LWG 3938</a> </td> <td>C ++ 23</td> <td>
<code>or_else</code>        trả lại std :: gọi (std :: forward &lt;f&gt; (f));<code>T</code>    khác<br/>        trả lại std :: gọi (std :: forward &lt;f&gt; (f), ** this);<span class="kw4">Vô hiệu</span>Và<code>E</code>}</td> <td>Không giống như với</td>
</tr>
</table> <h3 id="See_also">(chức năng)</h3> <table class="t-dsc-begin"> <tr class="t-dsc"> <td> <div><a href="transform_error" title="cpp/utility/expected/transform error"> <span class="t-lines"><span>chính nó nếu nó chứa một giá trị dự kiến; Nếu không, trả về kết quả của hàm đã cho trên giá trị bất ngờ</span></span></a></div> </td> <td>trả lại<code>expected</code>Transform_error<code>expected</code>chính nó nếu nó chứa một giá trị dự kiến; Nếu không, trả lại một<br/> <span class="t-mark">(chức năng thành viên công cộng)</span> </td>
</tr> </table> <div class="_attribution">
<p class="_attribution-p">© CPPReference.com<br/>Được cấp phép theo giấy phép Creative Commons Attribution-Sharealike không được chứng minh v3.0.<br/>
<a class="_attribution-link" href="https://en.cppreference.com/w/cpp/utility/expected/or_else">https://en.cppreference.com/w/cpp/utility/expected/or_else</a>
</p>
</div>
